### PR TITLE
Lazy load model docstring

### PR DESCRIPTION
**Pull request by dnephin**
Resolves the second point in #87

Based on https://github.com/prat0318/swagger-py/compare/Lazy_model_docstring

Added some unit tests for model building which replaced a couple of the integration tests.

I did some manual testing as well, to verify the ipython interactive behaviour works as expected, and that `create_model_docstring` is not in the cprofile for a regular non-interactive call.

### CODE REVIEW

File: tests/resource_model_test.py [hunk 1]
```diff
@@ -168,19 +161,6 @@ def test_nones_for_dates_on_model_types_creation(self):
         )
 
     @httpretty.activate
-    def test_success_on_model_types_instantiation(self):
```
Comment (prat0318) [inline tests/resource_model_test.py]: The test name might be misleading, but i think the test checks the `__eq__` method implementation of the model. Is the test redundant / unnecessary?

**Comment by prat0318** [comment]
lgtm.

File: tests/resource_model_test.py [hunk 2]
```diff
@@ -145,13 +145,6 @@ def register_urls(self):
     ################################################################
 
     @httpretty.activate
-    def test_success_on_model_types_creation(self):
```
Comment (prat0318) [inline tests/resource_model_test.py]: Not sure, if this test is redundant?

Comment (dnephin) [inline tests/resource_model_test.py]: The tests for `vars()`, `dir()` and types cover this I think

File: tests/resource_model_test.py (hunk 1 above)
Comment (dnephin) [inline tests/resource_model_test.py]: Yup, this is covered by line 37 in the new test

File: swaggerpy/swagger_model.py [hunk 3]
```diff
@@ -307,35 +299,35 @@ def load_json(resource_listing, http_client=None, processors=None):
...
-            swagger_type.get_swagger_types(props))
-    # Define a class variable to store all required fields
-    setattr(model_type, '_required', model.get('required'))
-    return model_type
+        __dir__=lambda self: props.keys(),
+        _flat_dict=lambda self: create_flat_dict(self),
+        _swagger_types=swagger_type.get_swagger_types(props),
+        _required=model.get('required'),
+    )
+    return type(name, (object,), methods)
```
Comment (analogue) [inline swaggerpy/swagger_model.py]: I think you need to call out in simple terms why we're doing this in the first place and the net effect: performance reasons and lazy docstring generation in REPL

**Comment by analogue** [comment]
Fix-n-ship!

File: swaggerpy/swagger_model.py (hunk 3 above)
Comment (dnephin) [inline swaggerpy/swagger_model.py]: Good call, will document this

**Pull request merged**